Can't find localization_data.json

Can't run search-wordpress-on-google.js test, because

> Error: Cannot find module '../../localization_data.json'

Where I can get an example for this file?
Thanks.

--- a/lib/flows/google-flow.js
+++ b/lib/flows/google-flow.js
@@ -33,7 +33,7 @@ export default class GoogleFlow {
 
 		this.GoogleAdPreviewTool.search( params.query );
 		var searchUrl = this.GoogleAdPreviewTool.getSearchPageUrl();
-		d.fulfill( new GoogleSearchPage( driver, data.wpcom_base_url ) );
+		d.fulfill( new GoogleSearchPage( driver, 'https://' + data.wpcom_base_url ) );
 		return d.promise;
 	}
 }

--- a/lib/flows/google-flow.js
+++ b/lib/flows/google-flow.js
@@ -7,7 +7,6 @@ import * as driverManager from '../driver-manager.js';
 import * as dataHelper from '../data-helper.js';
 
 import GoogleAdPreviewTool from '../pages/external/google-ad-preview-tool.js';
-import GoogleSearchPage from '../pages/external/google-search.js';
 
 export default class GoogleFlow {
 	constructor( driver, screenSize ) {
@@ -22,18 +21,13 @@ export default class GoogleFlow {
 	}
 
 	search( params, data ) {
-		var d = webdriver.promise.defer();
-		var driver = this.driver;
 		var screenSize = this.screenSize;
-		this.GoogleAdPreviewTool = new GoogleAdPreviewTool( driver, data.language, params.originating_location );
+		this.GoogleAdPreviewTool = new GoogleAdPreviewTool( this.driver, screenSize, params.domain, params.location, params.query );
 
-		driver.takeScreenshot().then( function( data ) {
+		this.driver.takeScreenshot().then( function( data ) {
 			// mediaHelper.writeScreenshot( data, data.language.toUpperCase() + '_' + screenSize + 'GOOGLE', true );
 		} );
 
-		this.GoogleAdPreviewTool.search( params.query );
-		var searchUrl = this.GoogleAdPreviewTool.getSearchPageUrl();
-		d.fulfill( new GoogleSearchPage( driver, 'https://' + data.wpcom_base_url ) );
-		return d.promise;
+		return this.GoogleAdPreviewTool.getSearchPageUrl();
 	}
 }

--- a/lib/pages/external/google-ad-preview-tool.js
+++ b/lib/pages/external/google-ad-preview-tool.js
@@ -1,83 +1,34 @@
 import webdriver, { By as by, until } from 'selenium-webdriver';
 import assert from 'assert';
 import BaseContainer from '../../base-container.js';
-import * as driverHelper from '../../driver-helper';
+import * as driverManager from '../../driver-manager';
 
 export default class GoogleAdPreviewTool extends BaseContainer {
-	constructor( driver, language, originating_location ) {
-		var url = "https://adwords.google.com/apt/anon/AdPreview";
-		super( driver, by.id( 'com.google.ads.apps.anonymousapt.GwtModule' ), true, url );
-		this.waitForPage();
-		if ( originating_location ) {
-			this.setLocation( originating_location );
+	constructor( driver, screenSize, domain, location, query ) {
+		if ( screenSize === 'mobile' ) {
+			screenSize = 30001;
+		} else if ( screenSize === 'tablet' ) {
+			screenSize = 30002;
+		} else {
+			screenSize = 30000;
 		}
-		this.setLanguage( language );
-		this.setComDomain();
-		this.checkURL();
-	}
-	setLocation( location ) {
-		var d = webdriver.promise.defer();
-		var selector = by.id( 'gwt-debug-diagnose-keywords-location-pillv2' );
-		this.driver.wait( until.elementLocated( selector ), this.explicitWaitMS, 'Could not locate the Location box' );
-		var element = this.driver.findElement( selector );
-		this.driver.wait( until.elementIsVisible( element ), this.explicitWaitMS, 'Could not see Location box' );
-		element.click();
 
-		var locationSearchbox = this.driver.findElement( by.id( 'gwt-debug-geo-search-box' ) );
-		locationSearchbox.sendKeys( location + "\n" );
-
-		selector = by.id( 'gwt-debug-geo-suggestions-pop-up' );
-		this.driver.wait( until.elementLocated( selector ), this.explicitWaitMS, 'Could not locate the suggestions dropdown' );
-		element = this.driver.findElement( selector );
-		this.driver.wait( until.elementIsVisible( element ), this.explicitWaitMS, 'Could not see suggestions dropdown' );
-
-		locationSearchbox.sendKeys( "\n" );
-		this.driver.findElement( by.css( 'body' ) ).click();
-		d.fulfill( true );
-		return d.promise;
-	}
-	setLanguage( language ) {
-		var d = webdriver.promise.defer();
-		this.driver.findElement( by.id( 'gwt-debug-diagnose-keywords-languages-widgetv2' ) ).click();
-		this.driver.findElement( by.xpath( "//div[contains(text(), '" + language + "')]" ) ).click();
-		d.fulfill( true );
-		return d.promise;
-	}
-	setComDomain( language ) {
-		var d = webdriver.promise.defer();
-		this.driver.findElement( by.id( 'gwt-debug-diagnose-keywords-domains-widgetv2' ) ).click();
-		this.driver.findElement( by.xpath( "//div[contains(text(), 'com')]" ) ).click();
-		d.fulfill( true );
-		return d.promise;
-	}
-	search( searchTerm ) {
-		var d = webdriver.promise.defer();
-		driverHelper.setWhenSettable( this.driver, by.id( 'gwt-debug-diagnostic-queryv2' ), searchTerm );
-		this.driver.findElement( by.id( 'gwt-debug-preview-ads-buttonv2-content' ) ).click();
-		d.fulfill( true );
-		return d.promise;
+		var url = "https://adwords.google.com/anon/AdPreview" +
+			"?lang=" + driverManager.currentLocale() +
+			"&loc=" + location +
+			"&device=" + screenSize +
+			"&st=" + escape( query ) +
+			"&domain=" + domain;
+		super( driver, by.css( '.adt-root' ), true, url );
+		this.waitForPage();
 	}
 
 	getSearchPageUrl() {
-		var d = webdriver.promise.defer();
-		var driver = this.driver;
-		var selector = by.name( 'resultsFramev2' );
+		var selector = by.css( 'iframe.iframe-preview' );
 		this.driver.wait( until.elementLocated( selector ), this.explicitWaitMS, 'Could not locate the search results' );
 		const iframe = this.driver.findElement( selector );
 		this.driver.wait( until.elementIsVisible( iframe ), this.explicitWaitMS, 'Could not see search results' );
 
-		iframe.getAttribute( 'src' ).then( ( src ) => {
-			driver.get( src );
-			return true;
-		} );
-
-		d.fulfill( true );
-		return d.promise;
-	}
-
-	checkURL() {
-		this.driver.getCurrentUrl().then( ( currentUrl ) => {
-			assert.equal( true, currentUrl.includes( 'google.com' ), `The current url: '${ currentUrl }' does not include 'google.com'` );
-		} );
+		return iframe.getAttribute( 'src' );
 	}
 }

--- a/lib/pages/external/google-ad-preview-tool.js
+++ b/lib/pages/external/google-ad-preview-tool.js
@@ -23,22 +23,11 @@ export default class GoogleAdPreviewTool extends BaseContainer {
 	}
 
 	getSearchPageUrl() {
-		var d = webdriver.promise.defer();
-		var driver = this.driver;
 		var selector = by.css( 'iframe.iframe-preview' );
 		this.driver.wait( until.elementLocated( selector ), this.explicitWaitMS, 'Could not locate the search results' );
 		const iframe = this.driver.findElement( selector );
 		this.driver.wait( until.elementIsVisible( iframe ), this.explicitWaitMS, 'Could not see search results' );
 
-		iframe.getAttribute( 'src' ).then( ( src ) => {
-			setTimeout( function() {
-				driver.get( src );
-			}, 5000 );
-
-			return true;
-		} );
-
-		d.fulfill( true );
-		return d.promise;
+		return iframe.getAttribute( 'src' );
 	}
 }

--- a/lib/pages/external/google-ad-preview-tool.js
+++ b/lib/pages/external/google-ad-preview-tool.js
@@ -1,5 +1,4 @@
 import webdriver, { By as by, until } from 'selenium-webdriver';
-import assert from 'assert';
 import BaseContainer from '../../base-container.js';
 import * as driverManager from '../../driver-manager';
 
@@ -24,11 +23,22 @@ export default class GoogleAdPreviewTool extends BaseContainer {
 	}
 
 	getSearchPageUrl() {
+		var d = webdriver.promise.defer();
+		var driver = this.driver;
 		var selector = by.css( 'iframe.iframe-preview' );
 		this.driver.wait( until.elementLocated( selector ), this.explicitWaitMS, 'Could not locate the search results' );
 		const iframe = this.driver.findElement( selector );
 		this.driver.wait( until.elementIsVisible( iframe ), this.explicitWaitMS, 'Could not see search results' );
 
-		return iframe.getAttribute( 'src' );
+		iframe.getAttribute( 'src' ).then( ( src ) => {
+			setTimeout( function() {
+				driver.get( src );
+			}, 5000 );
+
+			return true;
+		} );
+
+		d.fulfill( true );
+		return d.promise;
 	}
 }

--- a/lib/pages/external/google-search.js
+++ b/lib/pages/external/google-search.js
@@ -4,10 +4,12 @@ import BaseContainer from '../../base-container.js';
 
 export default class GoogleSearchPage extends BaseContainer {
 	constructor( driver, referenceUrl ) {
-		super( driver, by.css( 'li.ads-ad' ) );
-		this.waitForPage();
-		this.checkURL();
+		const frame = by.css( 'iframe.iframe-preview' );
+		super( driver, frame );
 		this.referenceUrl = referenceUrl;
+
+		const iframe = this.driver.findElement( frame );
+		driver.switchTo().frame( iframe );
 
 		const selector = by.xpath( '//li[@class="ads-ad"]//a[contains(@href, "' + referenceUrl +'")]' )
 		this.driver.wait( until.elementLocated( selector ), this.explicitWaitMS, 'Could not locate the ad link' );
@@ -42,11 +44,5 @@ export default class GoogleSearchPage extends BaseContainer {
 		} );
 
 		return d.promise;
-	}
-
-	checkURL() {
-		this.driver.getCurrentUrl().then( ( currentUrl ) => {
-			assert.equal( true, currentUrl.includes( 'google.' ), `The current url: '${ currentUrl }' does not include 'google.'` );
-		} );
 	}
 }

--- a/lib/pages/external/google-search.js
+++ b/lib/pages/external/google-search.js
@@ -4,11 +4,12 @@ import BaseContainer from '../../base-container.js';
 
 export default class GoogleSearchPage extends BaseContainer {
 	constructor( driver, referenceUrl ) {
-		super( driver, by.css( 'iframe li.ads-ad' ) );
+		super( driver, by.css( 'li.ads-ad' ) );
 		this.waitForPage();
+		this.checkURL();
 		this.referenceUrl = referenceUrl;
 
-		const selector = by.xpath( '//iframe//li[@class="ads-ad"]//a[contains(@href, "' + referenceUrl +'")]' )
+		const selector = by.xpath( '//li[@class="ads-ad"]//a[contains(@href, "' + referenceUrl +'")]' )
 		this.driver.wait( until.elementLocated( selector ), this.explicitWaitMS, 'Could not locate the ad link' );
 		this.adLink = this.driver.findElement( selector );
 		this.driver.wait( until.elementIsVisible( this.adLink ), this.explicitWaitMS, 'Could not see ad link' );
@@ -41,5 +42,11 @@ export default class GoogleSearchPage extends BaseContainer {
 		} );
 
 		return d.promise;
+	}
+
+	checkURL() {
+		this.driver.getCurrentUrl().then( ( currentUrl ) => {
+			assert.equal( true, currentUrl.includes( 'google.' ), `The current url: '${ currentUrl }' does not include 'google.'` );
+		} );
 	}
 }

--- a/lib/pages/external/google-search.js
+++ b/lib/pages/external/google-search.js
@@ -4,12 +4,11 @@ import BaseContainer from '../../base-container.js';
 
 export default class GoogleSearchPage extends BaseContainer {
 	constructor( driver, referenceUrl ) {
-		super( driver, by.css( 'li.ads-ad' ) );
+		super( driver, by.css( 'iframe li.ads-ad' ) );
 		this.waitForPage();
-		this.checkURL();
 		this.referenceUrl = referenceUrl;
 
-		const selector = by.xpath( '//li[@class="ads-ad"]//a[contains(@href, "' + referenceUrl +'")]' )
+		const selector = by.xpath( '//iframe//li[@class="ads-ad"]//a[contains(@href, "' + referenceUrl +'")]' )
 		this.driver.wait( until.elementLocated( selector ), this.explicitWaitMS, 'Could not locate the ad link' );
 		this.adLink = this.driver.findElement( selector );
 		this.driver.wait( until.elementIsVisible( this.adLink ), this.explicitWaitMS, 'Could not see ad link' );
@@ -42,11 +41,5 @@ export default class GoogleSearchPage extends BaseContainer {
 		} );
 
 		return d.promise;
-	}
-
-	checkURL() {
-		this.driver.getCurrentUrl().then( ( currentUrl ) => {
-			assert.equal( true, currentUrl.includes( 'google.' ), `The current url: '${ currentUrl }' does not include 'google.'` );
-		} );
 	}
 }

--- a/lib/pages/wp-home-page.js
+++ b/lib/pages/wp-home-page.js
@@ -36,9 +36,9 @@ export default class WPHomePage extends BaseContainer {
 		return d.promise;
 	}
 	checkURL( culture ) {
-		const target = culture ?
-			localizationData[ culture ].wpcom_base_url :
-			'wordpress.com';
+		const target = culture
+			? localizationData[ culture ].wpcom_base_url
+			: 'wordpress.com';
 		this.driver.getCurrentUrl().then( ( currentUrl ) => {
 			assert.equal( true, currentUrl.includes( target ), `The current url: '${ currentUrl }' does not include ${ target }` );
 		} );

--- a/lib/pages/wp-home-page.js
+++ b/lib/pages/wp-home-page.js
@@ -1,12 +1,13 @@
 import webdriver from 'selenium-webdriver';
 import assert from 'assert';
 import BaseContainer from '../base-container.js';
+import localizationData from '../../localization-data.json';
 
 const by = webdriver.By;
 
 export default class WPHomePage extends BaseContainer {
 	constructor( driver, { visit = false, culture = 'en', setRegularView = false } = {} ) {
-		super( driver, by.css( 'body' ), visit, 'https://' + culture + '.wordpress.com' );
+		super( driver, by.css( 'body' ), visit, 'https://wordpress.com/' );
 		if ( setRegularView === true ) {
 			this.setRegularView( culture );
 		}
@@ -17,7 +18,7 @@ export default class WPHomePage extends BaseContainer {
 		this.driver.manage().addCookie( {
 			name: 'wp_locale_test_group',
 			path: '/',
-			value: culture + '.wordpress.com'
+			value: localizationData[ culture ].wpcom_base_url
 		} );
 		this.driver.navigate().refresh();
 		this.waitForPage();
@@ -35,9 +36,9 @@ export default class WPHomePage extends BaseContainer {
 		return d.promise;
 	}
 	checkURL( culture ) {
-		const target = culture
-			? culture + '.wordpress.com'
-			: 'wordpress.com' ;
+		const target = culture ?
+			localizationData[ culture ].wpcom_base_url :
+			'wordpress.com';
 		this.driver.getCurrentUrl().then( ( currentUrl ) => {
 			assert.equal( true, currentUrl.includes( target ), `The current url: '${ currentUrl }' does not include ${ target }` );
 		} );

--- a/localization-data.json
+++ b/localization-data.json
@@ -1,0 +1,275 @@
+{
+	"ar": {
+		"language": "العربية",
+		"currency_symbol": "$",
+		"nux_landing_page_string": "خطوة",
+		"wpcom_landing_page_string": "دخول",
+		"wpcom_base_url": "ar.wordpress.com",
+
+		"google_searches": [
+			{
+				"query": "wordpress",
+				"originating_location": "القاهرة, محافظة القاهرة, مصر",
+				"originating_location_english": "Cairo, Cairo Governorate, Egypt"
+			}
+		]
+	},
+
+
+	"de": {
+		"language": "Deutsch",
+		"currency_symbol": "€",
+		"nux_landing_page_string": "Schritt",
+		"wpcom_landing_page_string": "Anmelden",
+		"wpcom_base_url": "de.wordpress.com",
+
+		"google_searches": [
+			{
+				"query": "wordpress",
+				"originating_location": "Berlin, Berlin, Germany"
+			}
+		]
+	},
+
+
+	"en": {
+		"language": "English",
+		"currency_symbol": "$",
+		"nux_landing_page_string": "Step",
+		"wpcom_landing_page_string": "Log In",
+		"wpcom_base_url": "wordpress.com",
+
+		"google_searches": [
+			{
+				"query": "wordpress",
+				"originating_location": "Brooklyn, New York, United States"
+			}
+		]
+	},
+
+
+	"es": {
+		"language": "español",
+		"currency_symbol": "€",
+		"nux_landing_page_string": "Paso",
+		"wpcom_landing_page_string": "Acceder",
+		"wpcom_base_url": "es.wordpress.com",
+
+		"google_searches": [
+			{
+				"query": "wordpress",
+				"originating_location": "Madrid, Community of Madrid, Spain"
+			}
+		]
+	},
+
+
+	"fr": {
+		"language": "Français",
+		"currency_symbol": "€",
+		"nux_landing_page_string": "Étape",
+		"wpcom_landing_page_string": "Connexion",
+		"wpcom_base_url": "fr.wordpress.com",
+
+		"google_searches": [
+			{
+				"query": "wordpress",
+				"originating_location": "Paris, Ile-de-France, France"
+			}
+		]
+	},
+
+	"he": {
+		"language": "עברית",
+		"currency_symbol": "$",
+		"nux_landing_page_string": "שלב",
+		"wpcom_landing_page_string": "התחבר",
+		"wpcom_base_url": "he.wordpress.com",
+
+		"google_searches": [
+			{
+				"query": "wordpress",
+				"originating_location": "ירושלים, מחוז ירושלים, ישראל",
+				"originating_location_english": "Jerusalem, Jerusalem District, Israel"
+			}
+		]
+	},
+
+
+	"id": {
+		"language": "Bahasa",
+		"currency_symbol": "$",
+		"nux_landing_page_string": "Langkah",
+		"wpcom_landing_page_string": "Log Masuk",
+		"wpcom_base_url": "id.wordpress.com",
+
+		"google_searches": [
+			{
+				"query": "wordpress",
+				"originating_location": "Jakarta, Special Capital Region of Jakarta, Indonesia"
+			}
+		]
+	},
+
+
+	"it": {
+		"language": "Italiano",
+		"currency_symbol": "€",
+		"nux_landing_page_string": "Passaggio",
+		"wpcom_landing_page_string": "Accedi",
+		"wpcom_base_url": "it.wordpress.com",
+
+		"google_searches": [
+			{
+				"query": "wordpress",
+				"originating_location": "Rome, Lazio, Italy"
+			}
+		]
+	},
+
+
+	"ja": {
+		"language": "日本語",
+		"currency_symbol": "¥",
+		"nux_landing_page_string": "ステップ",
+		"wpcom_landing_page_string": "ログイン",
+		"wpcom_base_url": "ja.wordpress.com",
+
+		"google_searches": [
+			{
+				"query": "wordpress",
+				"originating_location": "Tokyo, Japan"
+			}
+		]
+	},
+
+
+	"ko": {
+		"language": "한국어",
+		"currency_symbol": "$",
+		"nux_landing_page_string": "단계",
+		"wpcom_landing_page_string": "로그인",
+		"wpcom_base_url": "ko.wordpress.com",
+
+		"google_searches": [
+			{
+				"query": "wordpress",
+				"originating_location": "Seoul, Seoul, South Korea"
+			}
+		]
+	},
+
+
+	"nl": {
+		"language": "Nederlands",
+		"currency_symbol": "€",
+		"nux_landing_page_string": "Stap",
+		"wpcom_landing_page_string": "Inloggen",
+		"wpcom_base_url": "nl.wordpress.com",
+
+		"google_searches": [
+			{
+				"query": "wordpress",
+				"originating_location": "Amsterdam, North Holland, Netherlands"
+			}
+		]
+	},
+
+
+	"pt-br": {
+		"language": "Português",
+		"currency_symbol": "$",
+		"nux_landing_page_string": "Etapa",
+		"wpcom_landing_page_string": "Fazer login",
+		"wpcom_base_url": "br.wordpress.com",
+
+		"google_searches": [
+			{
+				"query": "wordpress",
+				"originating_location": "Rio de Janeiro, Rio de Janeiro, Brasil"
+			}
+		]
+	},
+
+
+	"ru": {
+		"language": "Русский",
+		"currency_symbol": "$",
+		"nux_landing_page_string": "Шаг",
+		"wpcom_landing_page_string": "Войти",
+		"wpcom_base_url": "ru.wordpress.com",
+
+		"google_searches": [
+			{
+				"query": "wordpress",
+				"originating_location": "Moscow, Moscow, Russia"
+			}
+		]
+	},
+
+
+	"sv": {
+		"language": "svenska",
+		"currency_symbol": "€",
+		"nux_landing_page_string": "Steg",
+		"wpcom_landing_page_string": "Logga in",
+		"wpcom_base_url": "sv.wordpress.com",
+
+		"google_searches": [
+			{
+				"query": "wordpress",
+				"originating_location": "Stockholm, Stockholms län, Sverige"
+			}
+		]
+	},
+
+
+	"tr": {
+		"language": "Türkçe",
+		"currency_symbol": "$",
+		"nux_landing_page_string": "başlayalım",
+		"wpcom_landing_page_string": "Giriş",
+		"wpcom_base_url": "tr.wordpress.com",
+
+		"google_searches": [
+			{
+				"query": "wordpress",
+				"originating_location": "Istanbul, Istanbul, Turkey"
+			}
+		]
+	},
+
+
+	"zh-cn": {
+		"language": "中文 (簡體)",
+		"currency_symbol": "$",
+		"nux_landing_page_string": "步",
+		"wpcom_landing_page_string": "登录",
+		"wpcom_base_url": "zh-cn.wordpress.com",
+
+		"google_searches": [
+			{
+				"query": "wordpress",
+				"originating_location": "Beijing, Beijing, China"
+			}
+		]
+	},
+
+
+	"zh-tw": {
+		"language": "中文 (繁體)",
+		"currency_symbol": "$",
+		"nux_landing_page_string": "步",
+		"wpcom_landing_page_string": "登入",
+		"wpcom_base_url": "zh-tw.wordpress.com",
+
+		"google_searches": [
+			{
+				"query": "wordpress",
+
+				"originating_location": "台北市, 台灣",
+				"originating_location_english": "Taipei City, Taiwan"
+			}
+		]
+	}
+}

--- a/localization-data.json
+++ b/localization-data.json
@@ -304,7 +304,7 @@
 				"domain": "www.google.com.tw",
 				"query": "wordpress",
 				"location": 9040379,
-				"originating_location_english": "Taipei, Taiwan"
+				"comment_location": "Taipei, Taiwan"
 			}
 		]
 	}

--- a/localization-data.json
+++ b/localization-data.json
@@ -214,7 +214,7 @@
 				"domain": "www.google.com.br",
 				"query": "wordpress",
 				"location": 1001655,
-				"comment_location": "Rio de Janeiro, Brasil"
+				"comment_location": "Rio de Janeiro, Brazil"
 			}
 		]
 	},

--- a/localization-data.json
+++ b/localization-data.json
@@ -8,9 +8,10 @@
 
 		"google_searches": [
 			{
+				"url": "www.google.com.ar",
 				"query": "wordpress",
-				"originating_location": "القاهرة, محافظة القاهرة, مصر",
-				"originating_location_english": "Cairo, Cairo Governorate, Egypt"
+				"location": 1005390,
+				"comment_location": "Cairo, Egypt"
 			}
 		]
 	},
@@ -25,8 +26,10 @@
 
 		"google_searches": [
 			{
+				"domain": "www.google.de",
 				"query": "wordpress",
-				"originating_location": "Berlin, Berlin, Germany"
+				"location": 20226,
+				"comment_location": "Berlin, Germany"
 			}
 		]
 	},
@@ -41,8 +44,16 @@
 
 		"google_searches": [
 			{
+				"domain": "www.google.com",
 				"query": "wordpress",
-				"originating_location": "Brooklyn, New York, United States"
+				"location": 1022762,
+				"comment_location": "Brooklyn, USA"
+			},
+			{
+				"domain": "www.google.co.uk",
+				"query": "wordpress",
+				"location": 1006886,
+				"comment_location": "London, UK"
 			}
 		]
 	},
@@ -57,8 +68,10 @@
 
 		"google_searches": [
 			{
+				"domain": "www.google.es",
 				"query": "wordpress",
-				"originating_location": "Madrid, Community of Madrid, Spain"
+				"location": 1005493,
+				"comment_location": "Madrid, Spain"
 			}
 		]
 	},
@@ -68,13 +81,15 @@
 		"language": "Français",
 		"currency_symbol": "€",
 		"nux_landing_page_string": "Étape",
-		"wpcom_landing_page_string": "Connexion",
+		"wpcom_landing_page_string": "Se Connecter",
 		"wpcom_base_url": "fr.wordpress.com",
 
 		"google_searches": [
 			{
+				"domain": "www.google.fr",
 				"query": "wordpress",
-				"originating_location": "Paris, Ile-de-France, France"
+				"location": 1006094,
+				"comment_location": "Paris, France"
 			}
 		]
 	},
@@ -88,9 +103,10 @@
 
 		"google_searches": [
 			{
+				"domain": "www.google.co.il",
 				"query": "wordpress",
-				"originating_location": "ירושלים, מחוז ירושלים, ישראל",
-				"originating_location_english": "Jerusalem, Jerusalem District, Israel"
+				"location": 1007978,
+				"comment_location": "Jerusalem, Israel"
 			}
 		]
 	},
@@ -105,8 +121,10 @@
 
 		"google_searches": [
 			{
+				"domain": "www.google.co.id",
 				"query": "wordpress",
-				"originating_location": "Jakarta, Special Capital Region of Jakarta, Indonesia"
+				"location": 1007716,
+				"comment_location": "Jakarta, Indonesia"
 			}
 		]
 	},
@@ -121,8 +139,10 @@
 
 		"google_searches": [
 			{
+				"domain": "www.google.it",
 				"query": "wordpress",
-				"originating_location": "Rome, Lazio, Italy"
+				"location": 1008736,
+				"comment_location": "Rome, Italy"
 			}
 		]
 	},
@@ -137,8 +157,10 @@
 
 		"google_searches": [
 			{
+				"domain": "www.google.co.jp",
 				"query": "wordpress",
-				"originating_location": "Tokyo, Japan"
+				"location": 1009317,
+				"comment_location": "Tokyo, Japan"
 			}
 		]
 	},
@@ -153,8 +175,10 @@
 
 		"google_searches": [
 			{
+				"domain": "www.google.co.kr",
 				"query": "wordpress",
-				"originating_location": "Seoul, Seoul, South Korea"
+				"location": 1009871,
+				"comment_location": "Seoul, South Korea"
 			}
 		]
 	},
@@ -169,8 +193,10 @@
 
 		"google_searches": [
 			{
+				"domain": "www.google.nl",
 				"query": "wordpress",
-				"originating_location": "Amsterdam, North Holland, Netherlands"
+				"location": 1010543,
+				"comment_location": "Amsterdam, Netherlands"
 			}
 		]
 	},
@@ -185,8 +211,10 @@
 
 		"google_searches": [
 			{
+				"domain": "www.google.com.br",
 				"query": "wordpress",
-				"originating_location": "Rio de Janeiro, Rio de Janeiro, Brasil"
+				"location": 1001655,
+				"comment_location": "Rio de Janeiro, Brasil"
 			}
 		]
 	},
@@ -201,8 +229,10 @@
 
 		"google_searches": [
 			{
+				"domain": "www.google.ru",
 				"query": "wordpress",
-				"originating_location": "Moscow, Moscow, Russia"
+				"location": 1011969,
+				"comment_location": "Moscow, Russia"
 			}
 		]
 	},
@@ -217,8 +247,10 @@
 
 		"google_searches": [
 			{
+				"domain": "www.google.se",
 				"query": "wordpress",
-				"originating_location": "Stockholm, Stockholms län, Sverige"
+				"location": 1012228,
+				"comment_location": "Stockholm, Sverige"
 			}
 		]
 	},
@@ -233,8 +265,10 @@
 
 		"google_searches": [
 			{
+				"domain": "www.google.com.tr",
 				"query": "wordpress",
-				"originating_location": "Istanbul, Istanbul, Turkey"
+				"location": 1012782,
+				"comment_location": "Istanbul, Turkey"
 			}
 		]
 	},
@@ -249,8 +283,10 @@
 
 		"google_searches": [
 			{
+				"domain": "www.google.com",
 				"query": "wordpress",
-				"originating_location": "Beijing, Beijing, China"
+				"location": 1003334,
+				"comment_location": "Beijing, Beijing, China"
 			}
 		]
 	},
@@ -265,10 +301,10 @@
 
 		"google_searches": [
 			{
+				"domain": "www.google.com.tw",
 				"query": "wordpress",
-
-				"originating_location": "台北市, 台灣",
-				"originating_location_english": "Taipei City, Taiwan"
+				"location": 9040379,
+				"originating_location_english": "Taipei, Taiwan"
 			}
 		]
 	}

--- a/localization-data.json
+++ b/localization-data.json
@@ -286,7 +286,7 @@
 				"domain": "www.google.com",
 				"query": "wordpress",
 				"location": 1003334,
-				"comment_location": "Beijing, Beijing, China"
+				"comment_location": "Beijing, China"
 			}
 		]
 	},

--- a/magellan-i18n.json
+++ b/magellan-i18n.json
@@ -3,9 +3,9 @@
     "specs-i18n/"
   ],
   "mocha_opts": "test/mocha.opts",
-  "mocha_args": "-R spec-xunit-reporter",
-  "max_test_attempts": 2,
-  "max_workers": 3,
+  "mocha_args": "-R spec-xunit-reporter -d",
+  "max_test_attempts": 1,
+  "max_workers": 1,
   "suiteTag": "i18n",
   "sauce": false,
   "browser": "chrome",

--- a/run.sh
+++ b/run.sh
@@ -96,7 +96,8 @@ while getopts ":a:Rpb:s:gjWCH:wl:cm:fiIUvxu:h" opt; do
       ;;
     i)
       NODE_CONFIG_ARGS+=$I18N_CONFIG
-      LOCALES="en,pt-BR,es,ja,fr,he"
+      LOCALES="en"
+      # LOCALES="en,pt-BR,es,ja,fr,he"
       export SCREENSHOTDIR="screenshots-i18n"
       MAGELLAN_CONFIG="magellan-i18n-nux.json"
       ;;
@@ -218,7 +219,7 @@ else # Not using multiple CircleCI containers, just queue up the tests in sequen
         for config in "${MAGELLAN_CONFIGS[@]}"; do
           if [ "$config" != "" ]; then
             CMD="env BROWSERSIZE=$size BROWSERLOCALE=$locale $MAGELLAN --mocha_args='$MOCHA_ARGS' --config='$config' --max_workers=$WORKERS"
-  
+
             eval $CMD
             RETURN+=$?
           fi

--- a/run.sh
+++ b/run.sh
@@ -218,7 +218,7 @@ else # Not using multiple CircleCI containers, just queue up the tests in sequen
         for config in "${MAGELLAN_CONFIGS[@]}"; do
           if [ "$config" != "" ]; then
             CMD="env BROWSERSIZE=$size BROWSERLOCALE=$locale $MAGELLAN --mocha_args='$MOCHA_ARGS' --config='$config' --max_workers=$WORKERS"
-
+  
             eval $CMD
             RETURN+=$?
           fi

--- a/run.sh
+++ b/run.sh
@@ -96,8 +96,7 @@ while getopts ":a:Rpb:s:gjWCH:wl:cm:fiIUvxu:h" opt; do
       ;;
     i)
       NODE_CONFIG_ARGS+=$I18N_CONFIG
-      LOCALES="en"
-      # LOCALES="en,pt-BR,es,ja,fr,he"
+      LOCALES="en,pt-BR,es,ja,fr,he"
       export SCREENSHOTDIR="screenshots-i18n"
       MAGELLAN_CONFIG="magellan-i18n-nux.json"
       ;;

--- a/run.sh
+++ b/run.sh
@@ -102,6 +102,7 @@ while getopts ":a:Rpb:s:gjWCH:wl:cm:fiIUvxu:h" opt; do
       ;;
     I)
       SCREENSIZES=desktop
+      WORKERS=1 # We need to be careful to take it slow with Google
       NODE_CONFIG_ARGS+=$I18N_CONFIG
       LOCALES="en,es,pt-br,de,fr,he,ja,it,nl,ru,tr,id,zh-cn,zh-tw,ko,ar,sv"
       MAGELLAN_CONFIG="magellan-i18n.json"

--- a/specs-i18n/search-wordpress-on-google.js
+++ b/specs-i18n/search-wordpress-on-google.js
@@ -3,7 +3,7 @@ import config from 'config';
 
 import * as driverManager from '../lib/driver-manager.js';
 
-import localization_data from '../../localization_data.json';
+import localization_data from '../localization-data.json';
 import GoogleFlow from '../lib/flows/google-flow.js';
 import LandingPage from '../lib/pages/landing-page.js';
 

--- a/specs-i18n/search-wordpress-on-google.js
+++ b/specs-i18n/search-wordpress-on-google.js
@@ -64,5 +64,8 @@ function doGoogleAdSearch( search_params ) {
 
 test_data.google_searches.forEach( function( search_params ) {
 	doGoogleAdSearch( search_params );
+	if ( process.env.CIRCLECI === 'true' ) {
+		sleep( 15000 );
+	}
 } );
 

--- a/specs-i18n/search-wordpress-on-google.js
+++ b/specs-i18n/search-wordpress-on-google.js
@@ -20,6 +20,10 @@ test.before( function() {
 	driver = driverManager.startBrowser();
 } );
 
+test.afterEach( function() {
+	this.sleep( 15000 );
+} );
+
 function doGoogleAdSearch( search_params ) {
 	var description = 'Search for "' + search_params.query + '" on Google from ' +
 		search_params.originating_location +

--- a/specs-i18n/search-wordpress-on-google.js
+++ b/specs-i18n/search-wordpress-on-google.js
@@ -68,8 +68,5 @@ function doGoogleAdSearch( search_params ) {
 
 test_data.google_searches.forEach( function( search_params ) {
 	doGoogleAdSearch( search_params );
-	if ( process.env.CIRCLECI === 'true' ) {
-		sleep( 15000 );
-	}
 } );
 

--- a/specs-i18n/search-wordpress-on-google.js
+++ b/specs-i18n/search-wordpress-on-google.js
@@ -23,7 +23,7 @@ test.before( function() {
 
 test.after( function( done ) {
 	// Wait between tests to not overload Google
-	var wait_seconds = 45;
+	var wait_seconds = 10;
 	this.timeout( ( wait_seconds + 2 ) * 1e3 );
 	setTimeout( done, wait_seconds * 1e3);
 } );
@@ -46,9 +46,9 @@ function doGoogleAdSearch( search_params ) {
 		test.it( `Google search contains our ad`, function() {
 			const googleFlow = new GoogleFlow( driver, 'desktop' );
 			const that = this;
-			googleFlow.search( search_params, test_data ).then( searchPage => {
+			googleFlow.search( search_params, test_data ).then( searchPageUrl => {
 				that.searchPage = new GoogleSearchPage( driver, 'https://' + test_data.wpcom_base_url );
-			})
+			});
 		} );
 
 		test.it( `Our landing page exists`, function() {
@@ -56,7 +56,7 @@ function doGoogleAdSearch( search_params ) {
 			if ( ! this.searchPage ) {
 				this.skip( 'Depends on previous test passing' );
 			}
-			this.searchPage.getAdUrl().then( function( url ) {
+			this.searchPage.getAdUrl().then( url => {
 				that.landingPage = new LandingPage( driver, url );
 			} );
 		} );

--- a/specs-i18n/search-wordpress-on-google.js
+++ b/specs-i18n/search-wordpress-on-google.js
@@ -21,7 +21,7 @@ test.before( function() {
 } );
 
 test.afterEach( function() {
-	this.sleep( 15000 );
+	this.timeout( 15000 );
 } );
 
 function doGoogleAdSearch( search_params ) {

--- a/specs-i18n/search-wordpress-on-google.js
+++ b/specs-i18n/search-wordpress-on-google.js
@@ -5,6 +5,7 @@ import * as driverManager from '../lib/driver-manager.js';
 
 import localization_data from '../localization-data.json';
 import GoogleFlow from '../lib/flows/google-flow.js';
+import GoogleSearchPage from '../lib/pages/external/google-search.js';
 import LandingPage from '../lib/pages/landing-page.js';
 
 const mochaTimeOut = config.get( 'mochaTimeoutMS' );
@@ -20,19 +21,19 @@ test.before( function() {
 	driver = driverManager.startBrowser();
 } );
 
-test.afterEach( function() {
-	this.timeout( 45000 );
+test.after( function( done ) {
+	// Wait between tests to not overload Google
+	var wait_seconds = 45;
+	this.timeout( ( wait_seconds + 2 ) * 1e3 );
+	setTimeout( done, wait_seconds * 1e3);
 } );
 
 function doGoogleAdSearch( search_params ) {
 	if ( locale === 'tr' ) {
 		this.skip( 'Currently no advertising in this locale' );
 	}
-	var description = 'Search for "' + search_params.query + '" on Google from ' +
-		search_params.originating_location +
-		( search_params.originating_location_english
-			? ' (' + search_params.originating_location_english + ')'
-			: '' );
+	var description = 'Search for "' + search_params.query + '" on ' + search_params.domain + ' from ' +
+		search_params.comment_location;
 
 	test.describe( description + ' @i18n', function() {
 		this.timeout( mochaTimeOut );
@@ -45,9 +46,9 @@ function doGoogleAdSearch( search_params ) {
 		test.it( `Google search contains our ad`, function() {
 			const googleFlow = new GoogleFlow( driver, 'desktop' );
 			const that = this;
-			return googleFlow.search( search_params, test_data ).then( searchPage => {
-				that.searchPage = searchPage;
-			} )
+			googleFlow.search( search_params, test_data ).then( searchPage => {
+				that.searchPage = new GoogleSearchPage( driver, 'https://' + test_data.wpcom_base_url );
+			})
 		} );
 
 		test.it( `Our landing page exists`, function() {

--- a/specs-i18n/search-wordpress-on-google.js
+++ b/specs-i18n/search-wordpress-on-google.js
@@ -21,7 +21,7 @@ test.before( function() {
 } );
 
 test.afterEach( function() {
-	this.timeout( 15000 );
+	this.timeout( 45000 );
 } );
 
 function doGoogleAdSearch( search_params ) {

--- a/specs-i18n/search-wordpress-on-google.js
+++ b/specs-i18n/search-wordpress-on-google.js
@@ -25,6 +25,9 @@ test.afterEach( function() {
 } );
 
 function doGoogleAdSearch( search_params ) {
+	if ( locale === 'tr' ) {
+		this.skip( 'Currently no advertising in this locale' );
+	}
 	var description = 'Search for "' + search_params.query + '" on Google from ' +
 		search_params.originating_location +
 		( search_params.originating_location_english

--- a/specs-i18n/search-wordpress-on-google.js
+++ b/specs-i18n/search-wordpress-on-google.js
@@ -35,7 +35,7 @@ function doGoogleAdSearch( search_params ) {
 	var description = 'Search for "' + search_params.query + '" on ' + search_params.domain + ' from ' +
 		search_params.comment_location;
 
-	test.describe( description + ' @i18n', function() {
+	test.describe( description + ' @i18n (' + locale + ')', function() {
 		this.timeout( mochaTimeOut );
 
 		test.beforeEach( function() {

